src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1067,9 +1067,27 @@ private bool CheckPropertyValueKind(SyntaxNode node, BoundExpression expr, BindV
             }
 
             return true;
+
+            bool reportUseSite(MethodSymbol accessor)
+            {
+                UseSiteInfo<AssemblySymbol> useSiteInfo = accessor.GetUseSiteInfo();
+                if (!object.Equals(useSiteInfo.DiagnosticInfo, propertySymbol.GetUseSiteInfo().DiagnosticInfo))
+                {
+                    if (diagnostics.Add(useSiteInfo, propertySyntax))
+                    {
+                        return true;
+                    }
+                }
```

Comment:
>} [](start = 16, length = 1)

Is it intentional that in this case we return false but do not add dependencies?  Why would that be correct?

Reply:
> Or does ```diagnostics.Add``` also add dependencies?

Yes, it adds both.

---
In reply to: [361552089](https://github.com/dotnet/roslyn/pull/40560#discussion_r361552089) [](ancestors = 361552089)